### ISSUE TITLE
Accuracy varies depending on the target labels form: dense or sparse

### ISSUE THREAD

**Issue by MariaMsu** [PyTorchLightning/metrics#554]
## 🐛 Bug

If you calculate the accuracy for the same prediction, but in one case you use a dense target, and in the other case a sparse one, you will get a different result. This is a very unexpected behavior.

### To Reproduce

```py
import torch
import torchmetrics

t_dense = torch.tensor([2])
t_sparse = torch.tensor([[0, 0, 1]])

pred = torch.tensor([[0.3, 0.3, 0.4]])

acc_dense = torchmetrics.Accuracy()
acc_dense.update(pred, t_dense)
print(f'dense accuracy: {acc_dense.compute()}')

acc_sparse = torchmetrics.Accuracy()
acc_sparse.update(pred, t_sparse)
print(f'sparse accuracy: {acc_sparse.compute()}')


>> dense accuracy: 1.0
>> sparse accuracy: 0.6666666865348816
```

### Expected behavior

The accuracy should be the same for the sparse and dense target

### Environment

- PyTorch Version: 1.9.1+cu102
- OS: Ubuntu 20.04.2 LTS (x86_64)
- How you installed PyTorch: pip
- Build command you used (if compiling from source):
- Python version: 3.7.11
- CUDA/cuDNN version: 10.2
- GPU models and configuration: GM108M [GeForce 930MX]

**Comment by SkafteNicki** [comment]
That is fully expected. If you look at the table on this page: https://torchmetrics.readthedocs.io/en/latest/references/modules.html#classification-metrics you will see that the first case corresponds to doing multi-class evaluation (dense case) and the other corresponds to doing multi-label evaluation (sparse case).

First case computation:
```python
pred = torch.tensor([[0.3, 0.3, 0.4]])
pred_max = pred.argmax(dim=-1)
acc = (pred_max == t_dense).mean() # 1.0
```
Second case:
```python
pred = torch.tensor([[0.3, 0.3, 0.4]])
pred_binary = (pred > 0.5).float()  # 0.5 is the default threshold value in this case
acc = (pred_binary == t_sparse).mean() # 0.66
```
